### PR TITLE
feat(dpp): initial RS DPP integration

### PR DESCRIPTION
**Pull request by antouhou**
Important considerations:
- Serialization through `js_sys::JSON::stringify`, because it seems like `serde-wasm-bindgen` doesn't play nicely with serde-json

This PR provides the groundwork for Rust DPP integration into the platform

## Issue being fixed or feature implemented

Add basic WASM bindings for the Rust DPP

## What was done?

Add binding and created a basic glue template for all JS tests

## How Has This Been Tested?

Converted `Identity` and `PublicKey` tests to use WASM DPP data structures instead

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wasm-dpp/lib/test/fixtures/getDashPayContractFixture.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const dashPaySchema = require('@dashevo/dashpay-contract/schema/dashpay.schema.json');
```
Comment (qrayven) [inline packages/wasm-dpp/lib/test/fixtures/getDashPayContractFixture.js:1]: I'm not sure if I get why we need all these fixtures implemented in JS . Shouldn't we just use the ones from rust dpp?

**Review by qrayven — COMMENTED** [review]
Great work!
In terms of conversion: 

In a perfect world, it would be good to assume, that the rust-dpp is the 'source of truth' and any wrapper built on top of it should allow everything that rust-dpp can do or more (phase space of wrapper should be a superset).

In the real world, I think using numbers is enough at the moment. But we should leave a note that JSbinding introduces additional constraints compared to the original rust implementation.

File: packages/wasm-dpp/lib/test/fixtures/getDashPayContractFixture.js (hunk 1 above)
Comment (antouhou) [reply packages/wasm-dpp/lib/test/fixtures/getDashPayContractFixture.js:1]: Yep, absolutely! It's just the old test, and I'm replacing things one by one. This PR is already pretty big, so I would love to merge this one first before we continue with other stuff

File: packages/wasm-dpp/test/unit/identity/Identity.spec.js [hunk 2]
```diff
@@ -1,24 +1,35 @@
-const generateRandomIdentifier = require('../../../lib/test/utils/generateRandomIdentifier');
+const { default: loadWasmDpp } = require('../../../dist');
+const generateRandomIdentifierAsync = require('../../../lib/test/utils/generateRandomIdentifierAsync');
 
-const IdentityPublicKey = require('../../../lib/identity/IdentityPublicKey');
-const Metadata = require('../../../lib/Metadata');
-const protocolVersion = require('../../../lib/version/protocolVersion');
+const lodashCloneDeep = require('lodash.clonedeep');
+const IdentityPublicKey = require('@dashevo/dpp/lib/identity/IdentityPublicKey');
```
Comment (shumkov) [inline packages/wasm-dpp/test/unit/identity/Identity.spec.js:5]: Rename JSIdentityPublicKey

File: packages/wasm-dpp/test/unit/identity/Identity.spec.js [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
   let hashMock;
   let encodeMock;
   let metadataFixture;
+  let Identity;
+  let Metadata;
+  let IdentityPublicKeyWasm;
 
-  beforeEach(function beforeEach() {
+  before(async () => {
+    ({ Identifier, Identity, Metadata, IdentityPublicKey: IdentityPublicKeyWasm } = await loadWasmDpp());
```
Comment (shumkov) [inline packages/wasm-dpp/test/unit/identity/Identity.spec.js:25]: Rename to `IdentityPublicKey`

File: packages/wasm-dpp/test/unit/identity/Identity.spec.js [hunk 4]
```diff
@@ -27,6 +38,7 @@ describe('Identity', () => {
           purpose: IdentityPublicKey.PURPOSES.AUTHENTICATION,
```
Comment (shumkov) [inline packages/wasm-dpp/test/unit/identity/Identity.spec.js:38]: Shouldn't we export constants from wasm as well?

File: packages/wasm-dpp/test/unit/identity/IdentityPublicKey.spec.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
-const IdentityPublicKey = require('../../../lib/identity/IdentityPublicKey');
-const EmptyPublicKeyDataError = require('../../../lib/identity/errors/EmptyPublicKeyDataError');
+const { default: loadWasmDpp } = require('@dashevo/wasm-dpp');
+const IdentityPublicKey = require('@dashevo/dpp/lib/identity/IdentityPublicKey');
```
Comment (shumkov) [inline packages/wasm-dpp/test/unit/identity/IdentityPublicKey.spec.js:2]: Rename to `JSIdentityPublicKey`

File: packages/wasm-dpp/test/unit/identity/IdentityPublicKey.spec.js [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
 
 describe('IdentityPublicKey', () => {
   let rawPublicKey;
   let publicKey;
+  let IdentityPublicKeyWasm;
```
Comment (shumkov) [inline packages/wasm-dpp/test/unit/identity/IdentityPublicKey.spec.js:7]: Rename to `IdentityPublicKey`

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/wasm-dpp/test/unit/identity/Identity.spec.js (hunk 4 above)
Comment (qrayven) [reply packages/wasm-dpp/test/unit/identity/Identity.spec.js:38]: I used exports from rs-dpp, but it seems impossible to keep `IdentityPublicKey` namespace for the constants.

File: packages/wasm-dpp/test/unit/identity/IdentityPublicKey.spec.js [hunk 7]
```diff
@@ -1,38 +1,32 @@
...
-  beforeEach(() => {
+  beforeEach(async () => {
     rawPublicKey = {
       id: 0,
-      type: IdentityPublicKey.TYPES.ECDSA_SECP256K1,
+      type: JSIdentityPublicKey.TYPES.ECDSA_SECP256K1,
       data: Buffer.from('AkVuTKyF3YgKLAQlLEtaUL2HTditwGILfWUVqjzYnIgH', 'base64'),
-      purpose: IdentityPublicKey.PURPOSES.AUTHENTICATION,
-      securityLevel: IdentityPublicKey.SECURITY_LEVELS.MASTER,
+      purpose: JSIdentityPublicKey.PURPOSES.AUTHENTICATION,
```
Comment (shumkov) [inline packages/wasm-dpp/test/unit/identity/IdentityPublicKey.spec.js]: Please move constants to wasm

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/wasm-dpp/src/mocks.rs [hunk 8]
```diff
@@ -0,0 +1,29 @@
+//! This module contains data structures that are left to be implemented
```
[the remark this answers is not in the document]
Comment (antouhou) [reply packages/wasm-dpp/src/mocks.rs:1]: This is a mock for DocumentTransition binding, which is currently not implemented just yet

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**